program/src/processor/update.rs

```diff
@@ -69,13 +69,14 @@ pub fn update_state(
     // write updated state if there was a change
```

Comment:
I cannot add a comment to the code that isn't part of the diff, but I think for consistency you might want to add the derivation check for `migration_state_info`?

Reply:
Also for code above, bear in mind these formatted prints eat up compute units if `RuleSet` gets big (saying from experience).  I've had a couple of these blow the default compute limit:
```      
msg!("new rule set provided");
msg!("rule set: {:?}", rule_set);
```

Reply:
Yeah, I'm leaving in some debug messages until prod launch because debugging errors is really painful. These particular ones can probably go away but don't worry too much about seeing some of them on `develop`.

Reply:
> I cannot add a comment to the code that isn't part of the diff, but I think for consistency you might want to add the derivation check for `migration_state_info`?

With the `is_zeroed` check added to `from_account_info` and the program owner check we are assured that any migration state accounts passed in are valid:

* is_zeroed ensures there is data present and therefore the account is initialized
* program owner ensures the program is owned by mpl-migration-validator

Only the owning program can write data, so this ensures this is a valid PDA created by our program. Once we know it's valid we can check that the authority is a signer and the same authority on the deserialized state account which validates that everything is as it should be.

Since I'm not particularly worried about compute in this program I could add the `migration_state_derived_from_mint` derivation check but then would need to require the mint account passed in. 

If I'm missing an attack vector here let me know and I can do that but I think it's covered as is.

Reply:
I just realized you're suggesting I do the same thing I do in `close` for consistency. I can add the derivation that checks the derivation using the mint from the deserialized state account. I can do this but don't think it is actually necessary.

Reply:
yeah it was more a consistency thing at first, but also originally the `is_zeroed` wasn't there.  But now the `is_zeroed` is there so we are pretty confident in the account.